deepinv/training/run_logger.py

```diff
@@ -0,0 +1,499 @@
+
+        :param images: Dictionary of images to log.
+        :param int step: Current training step.
+        :param int epoch: Current training epoch.
+        :param str phase: Training phase ('train', 'val', 'test').
+        """
+        pass
+
+    @abstractmethod
+    def load_from_checkpoint(self, checkpoint_dict: dict[str, Any]) -> None:
+        """
+        Resume the logger by restoring from a training checkpoint.
+
+        :param dict checkpoint_dict: Contains model weights, optimizer states, LR scheduler, etc.
+        """
+        pass
+
+    @abstractmethod
+    def prepare_checkpoint(self, checkpoint_dict: dict) -> dict:
+        """
+        Add logger specific information in training checkpoint.
+
+        :param dict checkpoint_dict: Contains model weights, optimizer states, LR scheduler, etc.
+        """
+        pass
+
+    @abstractmethod
+    def finish_run(self) -> None:
+        """
+        Finalize and close the training run.
+        """
+        pass
+
+
+class WandbLogger(RunLogger):
+    """
+    TODO
+    """
+
+    def __init__(
+        self,
+        log_dir: str = "logs",
+        project_name: str = "default_project",
+        run_name: str | None = None,
+        logging_mode: str = "online",
+        resume_id: str = None,
+    ) -> None:
+        """
+        TODO
+        """
+        self.proj_name = project_name
+        if run_name is None:
+            run_name = get_timestamp()
+        self.run_name = run_name
+        self.logging_mode = logging_mode
+        self.resume_id = resume_id
+
+        self.run_log_dir = Path(log_dir) / Path(self.project_name) / self.run_name
+
+    @classmethod
+    def get_wandb_setup(
+        cls,
+        wandb_save_dir: str,
+        wandb_proj_name: str,
+        wandb_run_name: str,
+        wandb_hp_config: dict[str, Any],
+        wandb_logging_mode: str = "online",
+        wandb_resume_id: str = None,
+    ) -> dict[str, Any]:
+        """
+        TODO
+        """
+
+        if (
+            wandb_resume_id is not None and wandb_logging_mode == "offline"
+        ):  # https://github.com/wandb/wandb/issues/2423
+            raise ValueError("Cannot resume wandb run with `wandb_logs_mode=offline`.")
+
+        if wandb_resume_id is not None:  # setting to resume a wandb run
+            wandb_setup = {
+                "dir": wandb_save_dir,
+                "mode": wandb_logging_mode,
+                "project": wandb_proj_name,
+                "id": wandb_resume_id,
+                "resume": "must",
+            }
+        else:  # setting to create a new wandb run
+            wandb_setup = {
+                "dir": wandb_save_dir,
+                "mode": wandb_logging_mode,
+                "project": wandb_proj_name,
+                "name": wandb_run_name,
+                "config": wandb_hp_config,
+            }
+        return wandb_setup
+
+    def init_logger(self, hyperparams: dict[str, Any] | None = None) -> None:
```

Comment:
What's the purpose of init_logger when the class has an init? I'm generally against "init" methods as ideally all init is done in the init.